### PR TITLE
[enhancement] CommandResults add indicator and return_results add list support

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
- [x] Ready

## Related Issues
step 1 in https://github.com/demisto/etc/issues/29016

## Description
Added to **return_results** support for list of results.

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [x] Documentation (updated `docstring`)

### CODE REVIEW

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -2760,6 +2760,17 @@ def return_results(results):
...
 
+    if isinstance(results, list):
+        if not results:
+            # demisto.results([]) is invalid
+            demisto.results({})
+            return
+
+        elif not isinstance(results[0], dict):  # list of dict can be handled by default case
+            for result in results:
+                return_results(result)
```
Comment (DeanArbel) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2771]: @anara123 comment: implement inline and not recursively.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 2]
```diff
@@ -2760,6 +2760,17 @@ def return_results(results):
         demisto.results(None)
         return
 
+    if isinstance(results, list):
+        if not results:
```
Comment (anara123) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2764]: no need for this validation

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -2760,6 +2760,17 @@ def return_results(results):
...
+            # demisto.results([]) is invalid
+            demisto.results({})
+            return
+
+        elif isinstance(results[0], CommandResults):  # other cases should be handled by default case
```
Comment (anara123) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2769]: check in single condition if results exists, is a list, has length > 0 and the first item is of type CommandResults

**Review by anara123 — CHANGES_REQUESTED** [review]

File: Packs/Base/ReleaseNotes/1_3_21.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### CommonServerPython
+- Added to **return_results** support for list of **CommandResults**.
```
Comment (amshamah419) [inline Packs/Base/ReleaseNotes/1_3_21.md:4]:
```suggestion
- Added support for lists of **CommandResults** objects to the **return_results** function.
```

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 5]
```diff
@@ -2760,6 +2760,11 @@ def return_results(results):
         demisto.results(None)
         return
 
+    if results and isinstance(results, list) and isinstance(results[0], CommandResults):
```
Comment (anara123) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2763]:
```suggestion
    if results and isinstance(results, list) and len(results) > 0 and isinstance(results[0], CommandResults):
```

**Comment by DeanArbel** [comment]
@amshamah419 Can you please review the new `Release Notes`?
Packs/AutoFocus/ReleaseNotes/1_1_7.md

File: Packs/AutoFocus/ReleaseNotes/1_1_7.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#### Integrations
+##### Palo Alto Networks AutoFocus v2
+- Updated reputation commands, **ip** **url** **domain** and **file**, to return a different result for each indicator.
```
Comment (amshamah419) [inline Packs/AutoFocus/ReleaseNotes/1_1_7.md:3]:
```suggestion
- Updated reputation the commands, **ip** **url** **domain** and **file**, to return a different result for each indicator. 
```

File: Packs/CyberTotal/ReleaseNotes/1_0_2.md [hunk 7]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### CyberTotal
+- Updated reputation commands, **ip** **url** **domain** and **file**, to return a different result for each indicator.
```
Comment (amshamah419) [inline Packs/CyberTotal/ReleaseNotes/1_0_2.md:4]:
```suggestion
- Updated the reputation commands, **ip** **url** **domain** and **file**, to return a different result for each indicator. 
```

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 8]
```diff
@@ -1419,21 +1406,14 @@ def search_domain_command(args):
...
-        raw_response=raw_response,
-        indicators=domain_indicator_list
-    )
+        command_results.append(CommandResults(
+            outputs_prefix='AutoFocus.Domain',
+            outputs_key_field='IndicatorValue',
+            outputs=autofocus_domain_output,
+            readable_output=md,
+            raw_response=raw_res,
+            indicators=[domain]
```
Comment (anara123) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:1415]: change this to be `indicator=domain`
leave the `indicators` arg in `__init__` too but use only `indicator`

**Review by anara123 — APPROVED** [review]
looks good. see my comments

**Comment by bakatzir** [comment]
@DeanArbel Can you please add a few lines regarding the changes in: https://xsoar.pan.dev/docs/integrations/code-conventions#commandresults ?

**Pull request merged**